lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java

```diff
@@ -1924,4 +1925,29 @@ public Object readSqlResultRecord(final ResultSet result) {
 		}
 
 	}
+
+    public List<SimplePageItem> getOrderedTopLevelPageItems(String siteId) {
+
+        // The unordered top level items
+        final List<SimplePageItem> tmpSiteItems = findItemsInSite(siteId);
+
+        final Optional<List<ToolConfiguration>> siteTools = getSiteTools(siteId);
+
+        if (!siteTools.isPresent()) {
+            return tmpSiteItems;
+        }
+
+        // build map of all pages, so we can see if any are left over
+        final Map<Long, SimplePage> pageMap = getSitePages(siteId)
+                .stream().collect(Collectors.toMap(SimplePage::getPageId, Function.identity()));
+
+        return siteTools.get().stream().map(t -> {
+
+            return tmpSiteItems
+                .stream()
+                .filter(spi -> pageMap.get(Long.valueOf(spi.getSakaiId())).getToolId().equals(t.getPageId()))
+                .findFirst().orElse(null);
```

Comment:
Is there an order here or should this be a findAny?

Reply:
Not sure. There should only ever be one of those, so I just used findFirst.